pkg/highlight/highlighter.go

```diff
@@ -77,6 +77,8 @@ type LineStates interface {
 	State(lineN int) State
```

Comment:
It's worth mentioning in the commit message that this commit actually fixes the race. (And perhaps right in the title, not in the body. "The new locking machanism" is *how* we do that, not *what* we do.)

[It's a shame that github doesn't allow adding review comments under commit messages]